SO-9115: Migrate to Envoy Gateway v1.6.0

riyaz-apica opened the pull request:
Replaced NGINX Ingress Controller with Envoy Gateway for modern ingress management.

Gateway Configuration:
- 4 listeners: HTTP (80), HTTPS (443), TCP-9999 (ingest), TCP-8081 (ML)
- Gateway API resources: Gateway, HTTPRoute, TCPRoute
- EnvoyProxy CRD for OCI Load Balancer configuration
- BackendTrafficPolicy and ClientTrafficPolicy for traffic management

OCI Load Balancer Configuration:
- Health check port: 10256
- externalTrafficPolicy: Cluster
- Flexible shape: 10-20 Mbps bandwidth
- 4 backend sets (optimized from 11)

Application Impact:
- All existing functionality preserved
- Access patterns unchanged (HTTP/HTTPS on 80/443, TCP on 9999/8081)
- Improved security with minimal external port exposure
- Better OCI integration and cost optimization 
 <div id='description'>
<a href="https://bito.ai#summarystart"></a>
<h3>Summary by Bito</h3>

<ul>

<li>Migrates ingress management from legacy NGINX Ingress Controller to Envoy Gateway v1.6.0, introducing improved security through new Custom Resource Definitions such as BackendTrafficPolicy, ClientTrafficPolicy, EnvoyProxy, Gateway, and GatewayClass, and new resources like HTTPRoute and TCPRoute for modern traffic management.</li>

<li>Removes legacy NGINX Ingress Controller configurations including ClusterRole, ClusterRoleBinding, ConfigMap, DaemonSet, Secret, Deployment, autoscaling, metrics, RBAC definitions, default backend components, and deprecated settings from Kubernetes Ingress Helm chart templates and CI directories.</li>

<li>Updates configurations across multiple deployment environments including AWS, Azure, K3s, MicroK8s, single node, small, standard, medium, large, and OCI setups, converting traditional ingress definitions into Gateway API format, replacing AWS ALB settings with nodePort-based configurations, and refining OCI load balancer settings and S3 storage configurations.</li>

<li>Enhances TLS conditional logic in the gateway YAML file to ensure secure HTTPS listeners are only created when TLS is enabled, improving security posture and supporting modern ingress practices.</li>

<li>Updates and refactors gateway configuration files and Helm charts to streamline configuration for OCI integration, enhance scalability, and improve cost efficiency while preserving existing functionality.</li>

<li>Overall summary: Migrates ingress management from legacy NGINX Ingress Controller to Envoy Gateway v1.6.0 by updating the main Helm chart with new version information, annotations, and dependency details; enhancing TLS logic; introducing new CRDs and resources; removing deprecated files and configurations; and improving security, scalability, cost efficiency, and OCI integration while aligning with modern ingress practices.</li>

</ul>

</div>

Commit:
SO-9115: Migrate to Envoy Gateway v1.6.0

Replaced NGINX Ingress Controller with Envoy Gateway for modern ingress management.

Gateway Configuration:
- 4 listeners: HTTP (80), HTTPS (443), TCP-9999 (ingest), TCP-8081 (ML)
- Gateway API resources: Gateway, HTTPRoute, TCPRoute
- EnvoyProxy CRD for OCI Load Balancer configuration
- BackendTrafficPolicy and ClientTrafficPolicy for traffic management

OCI Load Balancer Configuration:
- Health check port: 10256
- externalTrafficPolicy: Cluster
- Flexible shape: 10-20 Mbps bandwidth
- 4 backend sets (optimized from 11)

Application Impact:
- All existing functionality preserved
- Access patterns unchanged (HTTP/HTTPS on 80/443, TCP on 9999/8081)
- Improved security with minimal external port exposure
- Better OCI integration and cost optimization

leforsman commented on apica-ascent/values.aws.yaml:
```diff
@@ -187,30 +187,32 @@ s3gateway:
+# Gateway API configuration
+gateway:
+  tls:
+    enabled: false
```
@riyaz-apica does this need to be handled?

riyaz-apica replied:
https://github.com/ApicaSystem/apica-ascent-helm/pull/12/commits/af98e586cb9c5ef5db3d92d76420b7c690e48457

handled

Commit:
fix: Make HTTPS listener conditional on TLS being enabled

Only create HTTPS listener when TLS is enabled to ensure Gateway API compliance.
HTTPS listeners must have TLS configuration or Kubernetes will reject the Gateway.

esproul reviewed (changes requested):
I don't think we can rely on manual CRD creation for the onprem deployment model. This needs to be automated.

riyaz-apica commented:
> I don't think we can rely on manual CRD creation for the onprem deployment model. This needs to be automated.

https://apica.slack.com/archives/GG96MBVGX/p1764143373596519

Commit:
feat: Envoy Gateway v1.6.0 with multi-tenant support and NodePort conflict resolution

riyaz-apica committed:
Merge branch 'master' into SO-9115

leforsman commented on apica-ascent/README.md:
```diff
@@ -243,17 +336,17 @@ The deployment described above offers 30 days trial license. Email `license@apic
 
-Upload your APICA.IO deployment license
-- apicactl license set -l=license.jws
+# Upload your APICA.IO deployment license
```
This should say ascent and not apica.io, same for all comments in the readme.

riyaz-apica replied:
ok, you can edit the Readme after your work

leforsman commented on apica-ascent/charts/flash-coffee/templates/002-coffee-server-service.yaml:
```diff
@@ -1,7 +1,7 @@
 kind: Service
 metadata:
-  {{- if .Values.global.enableAwsAlb.enabled }}
+  {{- if .Values.global.nodePort.enabled }}
```
Here an AWS ALB gets configured if we specify nodeport. Is it correct?

riyaz-apica replied:
Parameter actually used for nodeport, its not a alb related parameter.

Commit:
fix: add missing container name to ssr-report-gen

esproul commented on apica-ascent/charts/logiq-flash/values.yaml:
```diff
@@ -176,10 +176,12 @@ affinity: {}
+  nodePort:
+    enabled: false
+    ports:
+      flash: null
+      flashMl: null
```
Should we preserve some default values for these ports, so that they can be turned on without needing to know the values? Are they the same as the ones used for ALB?

riyaz-apica replied:
No, these are for NodePort services. If you hard-code specific values, they may conflict with other environments since we are running everything in a single cluster. It’s better to keep them as null so Kubernetes can auto-assign an available NodePort for the service when needed.

esproul replied:
Understood, thanks.

esproul commented on apica-ascent/charts/thanos/values.yaml:
```diff
@@ -3741,6 +3741,8 @@ networkPolicy:
+     nodePort:
+       enabled: false
+       ports:
+         flash: null
+         flashMl: null
```
Should this be `flashThanosPort`, and should we preserve the default value?

riyaz-apica replied:
No, these are for NodePort services. If you hard-code specific values, they may conflict with other environments since we are running everything in a single cluster. It’s better to keep them as null so Kubernetes can auto-assign an available NodePort for the service when needed.

esproul replied:
I understand about default values but the change also turned `flashThanosPort` into the pair of `flash`/`flashMl` which seemed wrong to me, unless I misunderstand?

riyaz-apica replied:
https://github.com/ApicaSystem/apica-ascent-helm/commit/90b04c03614d887fe7c57bec93e711acd7c6d3b1

esproul commented on apica-ascent/templates/backend-traffic-policy.yaml:
```diff
@@ -0,0 +1,33 @@
+    kind: Gateway
+    name: {{ .Release.Name }}-gateway
+  
+  # Timeout configuration (from NGINX annotations)
```
I'm confused by the references to NGINX from here down. Are these still relevant for Envoy?

riyaz-apica replied:
I just added one comment to confirm that I’m using all the same configuration values from our previous NGINX Ingress Controller. The values are the same, but since this is now configured on the Envoy Gateway, we will remove the coment it accordingly in the future.

esproul replied:
OK, thanks.

Commit:
fix: Remove ALB annotations and add Thanos NodePort config
